### PR TITLE
AC-400: add concepts data providers, add visit form

### PR DESCRIPTION
**Pull request by Gelassen**
## Description of what I changed
The task was to make a data provider for concepts, that can operate both REST and cache sources. Feature for concepts that was cached already implemented few years ago and the same API was used for web request (SIC! apparently that concepts feature wasn't fully correct despite it passed code review, see details in ticket comments).

I make data provider for REST call and combine it with cache. I make a visit form as PoC of concepts autopopulate feature. I extend business logic of visit form to populate with all necessary data. Altought I faced with issue to find out way to submit entered data. It should be done to make visit note business valuable.  

Also I rename existing data providers from API postfix to Repository postfix as it more relevant to their current status and common naming convention. 
I introduced several business layers according to clean architecture principles in addition to MVP as current state of project needs it. 

## Issue I worked on
JIRA Issue: https://issues.openmrs.org/browse/AC-400

## Checklist: I completed these to help reviewers :)
- [x] My pull request only contains **ONE single commit**

- [ ] I have **added tests** to cover my changes. (If you refactored
existing code that was well tested you do not have to add tests)
(I noticed previous contributors stops cover changes with tests. Does it link to some extra work should be done to mock some classes?)

- [ ] All new and existing **tests passed**.
(Half of the tests fails. By looking through errors like this "Reason: cannot find android.support.v4.util.LruCache" seems they reffred to work of previous contributor. I think it is a good manners to fix them anyway, but let me do PR first to submit existing work to community as I contribute spare time to this project which I might lack of it at any time)

- [x] My pull request is **based on the latest changes** of the master branch.

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,6 +1,6 @@
-#Sun Feb 24 23:38:58 IST 2019
+#Wed May 08 11:26:14 MSK 2019
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dist
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-4.10.1-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-5.1.1-all.zip
```
[the remark this answers is not in the document]
Comment (Gelassen) [reply gradle/wrapper/gradle-wrapper.properties]: I double checked - yes, we need. Moreover PR is ready by another member, it is ready for merge. Could you please merge it? 

https://github.com/openmrs/openmrs-contrib-android-client/pull/587

[the remark this answers is not in the document]
Comment (Gelassen) [reply gradle/wrapper/gradle-wrapper.properties]: The lastest commit in master wasn't in the PR. I did rebase and push to server

[the remark this answers is not in the document]
Comment (Gelassen) [reply gradle/wrapper/gradle-wrapper.properties]: I did the PR for AC-608
https://github.com/openmrs/openmrs-contrib-android-client/pull/589

**Comment by Gelassen** [comment]
@f4ww4z, please confirm PR is ready for merge. Thank you.

**Comment by f4ww4z** [comment]
Yes @Gelassen , I will need @deepak140596 's response for #587 .

**Comment by Gelassen** [comment]
@f4ww4z , it seems all PRs are ready for merge, Could you please merge them?

**Comment by dkayiwa** [comment]
@Gelassen did you see the merge conflicts?

**Comment by f4ww4z** [comment]
@Gelassen Ping

**Comment by Gelassen** [comment]
Was out of work for severak weeks, I will be back to work next week

**Comment by f4ww4z** [comment]
@Gelassen work on #589 first before this PR.

**Pull request closed without merging**